src/core/src/OmniWebMapServiceRasterOverlay.cpp

```diff
@@ -0,0 +1,102 @@
+    : OmniRasterOverlay(pContext, path) {
+    reload();
+}
+
+CesiumRasterOverlays::RasterOverlay* OmniWebMapServiceRasterOverlay::getRasterOverlay() const {
+    return _pWebMapServiceRasterOverlay.get();
+}
+
+std::string OmniWebMapServiceRasterOverlay::getBaseUrl() const {
+    auto cesiumWebMapServiceRasterOverlay =
+        UsdUtil::getCesiumWebMapServiceRasterOverlay(_pContext->getUsdStage(), _path);
+    std::string baseUrl;
+    cesiumWebMapServiceRasterOverlay.GetBaseUrlAttr().Get(&baseUrl);
+    return baseUrl;
+}
+
+int OmniWebMapServiceRasterOverlay::getMinimumLevel() const {
+    auto cesiumWebMapServiceRasterOverlay =
+        UsdUtil::getCesiumWebMapServiceRasterOverlay(_pContext->getUsdStage(), _path);
+    int val;
+    cesiumWebMapServiceRasterOverlay.GetMinimumLevelAttr().Get(&val);
+    return val;
+}
+
+int OmniWebMapServiceRasterOverlay::getMaximumLevel() const {
+    auto cesiumWebMapServiceRasterOverlay =
+        UsdUtil::getCesiumWebMapServiceRasterOverlay(_pContext->getUsdStage(), _path);
+    int val;
+    cesiumWebMapServiceRasterOverlay.GetMaximumLevelAttr().Get(&val);
+    return val;
+}
+
+int OmniWebMapServiceRasterOverlay::getTileWidth() const {
+    auto cesiumWebMapServiceRasterOverlay =
+        UsdUtil::getCesiumWebMapServiceRasterOverlay(_pContext->getUsdStage(), _path);
+    int val;
+    cesiumWebMapServiceRasterOverlay.GetTileWidthAttr().Get(&val);
+    return val;
+}
+
+int OmniWebMapServiceRasterOverlay::getTileHeight() const {
+    auto cesiumWebMapServiceRasterOverlay =
+        UsdUtil::getCesiumWebMapServiceRasterOverlay(_pContext->getUsdStage(), _path);
+    int val;
+    cesiumWebMapServiceRasterOverlay.GetTileHeightAttr().Get(&val);
+    return val;
+}
+
+std::string OmniWebMapServiceRasterOverlay::getLayers() const {
+    auto cesiumWebMapServiceRasterOverlay =
+        UsdUtil::getCesiumWebMapServiceRasterOverlay(_pContext->getUsdStage(), _path);
+    std::string val;
+    cesiumWebMapServiceRasterOverlay.GetLayersAttr().Get(&val);
+    return val;
+}
+
+void OmniWebMapServiceRasterOverlay::reload() {
+    const auto rasterOverlayName = UsdUtil::getName(_pContext->getUsdStage(), _path);
+
+    CesiumRasterOverlays::RasterOverlayOptions options;
+    options.showCreditsOnScreen = getShowCreditsOnScreen();
+    // options.ktx2TranscodeTargets = GltfUtil::getKtx2TranscodeTargets();
+
+    options.loadErrorCallback = [this](const CesiumRasterOverlays::RasterOverlayLoadFailureDetails& error) {
+        _pContext->getLogger()->error(error.message);
+    };
+
+    CesiumRasterOverlays::WebMapServiceRasterOverlayOptions wmsOptions;
+    int minimumLevel = getMinimumLevel();
+    int maximumLevel = getMaximumLevel();
+    int tileWidth = getTileWidth();
+    int tileHeight = getTileHeight();
+    std::string layers = getLayers();
+
+    if (maximumLevel > minimumLevel) {
+        wmsOptions.minimumLevel = minimumLevel;
+        wmsOptions.maximumLevel = maximumLevel;
+    }
```

Comment:
This doesn't look right...

Reply:
Actually it's fine. But I wonder if a warning should be logged if `maximumLevel < minimumLevel`

Reply:
That's lifted from Unreal. I also thought about logging, then thought ideally it should be at the Python UI level. Users would have immediate and clear feedback. We should be able to add checking events into the DataModel when the values are changed. (The initial PR of tileset clipping did this.) I didn't want to crack that open in this PR, though. Now that I think about it, it shouldn't be hard. Let's try it at the Python level.

Reply:
This check was moved to the UI layer:
https://github.com/CesiumGS/cesium-omniverse/assets/1040194/3a80da3c-d165-4085-bea1-a33fe262601e